app/controllers/diagnoses_controller.rb

```diff
@@ -14,7 +14,9 @@ def processed
   end
 
   def new
-    @diagnosis = DiagnosisCreation.new_diagnosis(Solicitation.find_by(id: params[:solicitation]))
+    solicitation = Solicitation.find_by(id: params[:solicitation])
+    @diagnosis = DiagnosisCreation.new_diagnosis(solicitation)
+    @diagnoses = Solicitation.where(email: solicitation&.email).map(&:diagnoses).flatten
```

Comment:
On peut pas faire tout simplement un truc du genre:
```suggestion
    @diagnoses = Diagnosis.joins(:solicitation).where(solicitations: {email: solicitation.email})
```
?

Reply:
Ah, si on commence à faire ça, on peut ajouter un index sur `solicitations.email`. Ça peut servir :)

Reply:
J'avoue que je ne pense pas à raisonner avec des `joins`